sessions-core/src/main/java/com/adyen/checkout/sessions/api/SessionSetupService.kt

```diff
@@ -8,36 +8,43 @@
 
 package com.adyen.checkout.sessions.api
 
-import com.adyen.checkout.core.api.Connection
+import com.adyen.checkout.core.api.ConnectionHttpClient.Companion.CONTENT_TYPE_JSON_HEADER
 import com.adyen.checkout.core.api.Environment
+import com.adyen.checkout.core.api.HttpClientFactory
 import com.adyen.checkout.core.log.LogUtil
 import com.adyen.checkout.core.log.Logger
 import com.adyen.checkout.core.model.toStringPretty
 import com.adyen.checkout.sessions.model.setup.SessionSetupRequest
 import com.adyen.checkout.sessions.model.setup.SessionSetupResponse
-import java.io.IOException
-import org.json.JSONException
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
 import org.json.JSONObject
 
 private val TAG = LogUtil.getTag()
-private const val ENDPOINT = "v1/sessions/"
-
-class SessionSetupConnection(
-    private val request: SessionSetupRequest,
-    environment: Environment,
-    sessionId: String,
-    clientKey: String
-) : Connection<SessionSetupResponse>(
-    "${environment.baseUrl}$ENDPOINT$sessionId/setup?clientKey=$clientKey"
+
+internal class SessionSetupService(
+    private val environment: Environment
 ) {
-    @Throws(IOException::class, JSONException::class)
-    override fun call(): SessionSetupResponse {
-        Logger.v(TAG, "call - $url")
+
+    suspend fun setupSession(
```

Comment:
Just a thought for the next refactor. Do we want every API call to be just a method inside a Service class or to have its own separate class?
- For the first option I guess we keep what we have now but merge some classes together. For example `SessionBalanceService - SessionCancelOrderService - SessionCreateOrderService - SessionDetailsService - SessionPaymentsService - SessionSetupService` should all just be `SessionSevice`.
- In addition, if each API has its own a separate class, we could maybe make things more generic, like every API should provide its URL, method type, request and response type. This way Service classes can call any API the same way basically.
WDYT?

Reply:
Went with the first option in the next refactor. Made a method there to generify making requests, but we can take that a step further (will explain in that PR)